server/webhook_integration_test.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestSendToWebhook(t *testing.T) {
+	validPayload := &Payload{Username: "mattermod", Text: "test"}
+
+	mattermost := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "ok")
+	}))
+	defer mattermost.Close()
+
+	r, err := s.sendToWebhook(mattermost.URL, validPayload)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+	got := r.StatusCode
+	if got != http.StatusOK {
+		t.Errorf("got: %v, want: %v", got, http.StatusOK)
+	}
+	defer r.Body.Close()
+}
+
+func TestSendToWebhookUsernameNotSet(t *testing.T) {
+	invalidPayload := &Payload{Text: "test"}
+	mattermost := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	}))
+	defer mattermost.Close()
+
+	wantError := "username not set in webhook payload"
```

Comment:
String matching in errors is brittle and not recommended. We can name the error as a variable if we want to verify that. Or had a separate type like 'webhookValidationError' with field set to Username, Text etc. Up to you.

Same applies to the test below too.